frontend/src/entities/dataset/dataset.actions.tsx

```diff
@@ -58,15 +60,102 @@ function DatasetActions (props?: any) {
+export async function fileHandler (arrayOfFiles: File[], manageMode: DatasetBrowserManageMode | undefined, notificationService: any, {state, setState, updateDatasetContext, setDisableUpload,}: Pick<UploadButtonProperties, 'state' | 'setState' | 'updateDatasetContext' | 'setDisableUpload'>) {
+    const filesToUpload = arrayOfFiles.map((file: File): DatasetResourceObject => ({
+        bucket: '',
+        type: 'object',
+        key: `${state.datasetContext?.location || ''}${file.webkitRelativePath ? file.webkitRelativePath : file.name}`,
+        size: file.size,
+        file,
+        name: `${file.webkitRelativePath ? file.webkitRelativePath : file.name}`,
+    }));
+
+    switch (manageMode) {
+        case DatasetBrowserManageMode.Create:
+            setState({
+                type: DatasetActionType.State,
+                payload: {
+                    items: [...state.items, ...filesToUpload]
+                }
+            });
+            break;
+        case DatasetBrowserManageMode.Edit:
+            if (state.datasetContext) {
+                const filteringTextPrefix = prefixForPath(state.filteringText);
+                if (filteringTextPrefix) {
+                    filesToUpload.forEach((file) => {
+                        file.key = [filteringTextPrefix, file.key].join('');
+                    });
+                }
+                    
+                setDisableUpload(true);
+                // ensure cast to DatasetContext is valid
+                if (state.datasetContext.name && state.datasetContext.type) {
+                    await uploadResources(state.datasetContext as DatasetContext, filesToUpload, notificationService);
+                }
+                setDisableUpload(false);
+
+                // if the filter contains a prefix append that to the Location
+                const effectiveContext = {...state.datasetContext, location: [state.datasetContext.location, filteringTextPrefix].join('')};
+                updateDatasetContext(effectiveContext, '', false);
+            }
+            break;
+    }
+}
+
+type UploadButtonProperties = {
+    isFileUpload: boolean;
+    children: string;
+    state: Pick<DatasetBrowserState, 'selectedItems' | 'items' | 'datasetContext' | 'manageMode' | 'filteringText'>;
+    setState: ReduxDispatch<DatasetBrowserAction> | ReactDispatch<DatasetBrowserAction>;
+    updateDatasetContext: UpdateDatasetContextFunction;
+    disableUpload: boolean;
+    setDisableUpload: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const UploadButton = ({state, setState, updateDatasetContext, isFileUpload, disableUpload, setDisableUpload, children}: UploadButtonProperties): React.ReactNode => {
     const dispatch = useAppDispatch();
     const notificationService = NotificationService(dispatch);
+    const {manageMode} = state;
+    const uploadFile: RefObject<HTMLInputElement> = useRef(null);
+
+    
+
+    return <>
```

Comment:
What's the `<>` doing here?

Reply:
Its a fragment so that I don't have to have a div or any "parent" component wrapping these.